src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs

```diff
@@ -331,6 +329,11 @@ private void FireProcessingQueueEmpty()
 
         public Block? Process(Block suggestedBlock, ProcessingOptions options, IBlockTracer tracer)
         {
+            if (_options.LogProducedBlocks != DumpOptions.None && options == ProcessingOptions.ProducingBlock)
+            {
+                TraceProducedBlock(suggestedBlock, options);
+            }
+            
```

Comment:
Wouldn't it be better to change that in OneTimeChainProcessor, and instead of adding additional processing, just amend tracer, you can make a complex one easily.